sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ForLoopIncrementSignCheck.cs

```diff
@@ -34,25 +34,206 @@ namespace SonarAnalyzer.Rules.CSharp
+                    }
+                    else if (incrementorData.Operation == ArithmeticOperation.Substraction &&
+                             condition == Condition.Less)
+                    {
+                        c.ReportDiagnosticWhenActive(Diagnostic.Create(rule, forNode.Condition.GetLocation(), incrementorData.IdentifierName, "dec"));
                     }
                 },
                 SyntaxKind.ForStatement);
         }
+
+        private IncrementData GetIncrementData(ExpressionSyntax incrementor)
+        {
+            var identifierName = string.Empty;
+            var opp = ArithmeticOperation.None;
+
+            var incrementorKind = incrementor.Kind();
+
+            switch (incrementorKind)
+            {
+                case SyntaxKind.PreDecrementExpression:
+                case SyntaxKind.PreIncrementExpression:
+                case SyntaxKind.PostDecrementExpression:
+                case SyntaxKind.PostIncrementExpression:
+                    if (GetOperandFromUnaryExpression(incrementor) is IdentifierNameSyntax operand)
+                    {
+                        identifierName = operand.Identifier.ValueText;
+                        opp =  incrementorToOperation.GetValueOrDefault(incrementorKind, ArithmeticOperation.None);
+                    }
+                    break;
+
+                case SyntaxKind.SubtractAssignmentExpression:
+                case SyntaxKind.AddAssignmentExpression:
+                    if (incrementor is AssignmentExpressionSyntax add && add.Left is IdentifierNameSyntax leftId &&
+                        add.Right.IsKind(SyntaxKind.NumericLiteralExpression))
+                    {
+                        identifierName = leftId.Identifier.ValueText;
+                        opp = incrementorToOperation.GetValueOrDefault(incrementorKind, ArithmeticOperation.None);
+                    }
+                    break;
+
+                case SyntaxKind.SimpleAssignmentExpression:
+                    if (incrementor is AssignmentExpressionSyntax simpleAssignment &&
+                        simpleAssignment.Left is IdentifierNameSyntax simpleAssignmentId &&
+                        simpleAssignment.Right is BinaryExpressionSyntax right &&
+                        IsVariableAndLiteralBinaryExpression(right, simpleAssignmentId.Identifier.ValueText))
+                    {
+                        identifierName = simpleAssignmentId.Identifier.ValueText;
+                        opp = incrementorToOperation.GetValueOrDefault(right.Kind(), ArithmeticOperation.None);
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+
+            return new IncrementData(identifierName, opp);
+        }
+
+        private Condition GetCondition(BinaryExpressionSyntax conditionSyntax, string identifierName)
```

Comment:
it would be useful to add a comment specifying that this method treats conditions on both sides of the binary expression (e.g. `x < 1` and `1 > x` are the same thing, and this considers these cases)

Reply:
Done